src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpCheck.java

```diff
@@ -627,12 +627,10 @@ else if (!illegalPattern && matchCount == 0) {
     /**
      * Check if we can stop validation.
      *
-     * @param ignore flag
      * @return true is we can continue
      */
-    private boolean canContinueValidation(boolean ignore) {
-        return errorCount <= errorLimit - 1
-                && (ignore || illegalPattern || checkForDuplicates);
+    private boolean canContinueValidation() {
+        return errorCount <= errorLimit - 1;
```

Comment:
as we update code, we need to provide regression testing report on all projects.
Please make sure that config that will choose will use updated code.

Reply:
Regression report and code logic can be found here
#7983